test/system/src/main/java/io/pravega/test/system/framework/services/kubernetes/K8SequentialExecutor.java

```diff
@@ -126,33 +127,26 @@ private void verifyPravegaPodRestart(Map<String, V1ContainerStatus> podStatusBef
-                        .build())
-                .addNewContainer()
-                .withName(podName) // container name is same as that of the pod.
-                .withImage(TEST_POD_IMAGE)
-                .withImagePullPolicy("IfNotPresent")
-                .withCommand("/bin/sh")
-                .withArgs("-c", "java" +
-                        getArgs() +
-                        " -cp /data/test-collection.jar io.pravega.test.system.SingleJUnitTestRunner " + className + "#" + methodName /*+ " > server.log 2>&1 */ + "; exit $?")
-                .withVolumeMounts(new V1VolumeMountBuilder().withMountPath("/data").withName("task-pv-storage").build())
-                .endContainer()
-                .withRestartPolicy("Never")
-                .endSpec().build();
+        V1Pod pod =  new V1Pod()
+                .metadata(new V1ObjectMeta().name(podName).namespace(NAMESPACE).labels(ImmutableMap.of("POD_NAME", podName, "app", APP)))
+                .spec( new V1PodSpec().serviceAccountName(SERVICE_ACCOUNT).automountServiceAccountToken(true)
+                        .volumes(ImmutableList.of(new V1Volume().name("task-pv-storage")
+                                .persistentVolumeClaim(new V1PersistentVolumeClaimVolumeSource().claimName("task-pv-claim"))
+                        ))
+                        .containers( ImmutableList.of( new V1Container()
+                                .name(podName) // container name is same as that of the pod.
+                                .image(TEST_POD_IMAGE)
+                                .imagePullPolicy("IfNotPresent")
+                                .command(ImmutableList.of("/bin/sh"))
+                                .args( ImmutableList.of("-c", "java" +
+                                        getArgs() +
+                                        " -cp /data/test-collection.jar io.pravega.test.system.SingleJUnitTestRunner " + className + "#" + methodName /*+ " > server.log 2>&1 */ + "; exit $?"))
+                                .volumeMounts(ImmutableList.of(new V1VolumeMount().mountPath("/data").name("task-pv-storage")))
+                        ))
+                        .restartPolicy("Never"));
         if (Utils.TLS_AND_AUTH_ENABLED) {
-            pod  = new V1PodBuilder(pod).editSpec().withVolumes(new V1VolumeBuilder().withName("tls-certs")
-                                                  .withSecret(new V1SecretVolumeSourceBuilder().withSecretName(Utils.TLS_SECRET_NAME).build())
-                                                  .build())
-                .editContainer(0)
-                .withVolumeMounts(new V1VolumeMountBuilder().withMountPath(Utils.TLS_MOUNT_PATH).withName("tls-secret").build())
-                .endContainer()
-                .endSpec()
-                .build();
+            pod.getSpec().getVolumes().add(new V1Volume().name("tls-cert").secret(new V1SecretVolumeSource().defaultMode(420).secretName(Utils.TLS_SECRET_NAME)));
```

Comment:
is defaultMode a must now? (dont see it before)

Reply:
Also, what does `420` mean? Would it be better to define a constant with that value with a name/comments that explains what it is?

Reply:
Tried without the defaultMode. its optional the build is green even without that.